packages/mdl-menu/simple/foundation.js

```diff
@@ -299,13 +306,57 @@ export default class MDLSimpleMenuFoundation extends MDLFoundation {
     }, numbers.SELECTED_TRIGGER_DELAY);
   }
 
-  // Open the menu. Optionally focus on provided item.
+  autoPosition_() {
+    if (!this.adapter_.hasAnchor()) {
+      return;
+    }
+
+    // Defaults: open from the top left.
+    let vertical = 'top';
+    let horizontal = 'left';
+
+    const anchor = this.adapter_.getAnchorDimensions();
+    const windowDimensions = this.adapter_.getWindowDimensions();
+
+    // Would the preferred position go beyond the vertical bounds of the window?
+    const topOverflow = anchor.top + this.dimensions_.height - windowDimensions.height;
```

Comment:
I could do that, but then I would lose the ability to determine for very large menus which option is "less bad" in terms of overflow (see e.g. line 325, 337 or 340).